app/src/main/java/com/example/ohthmhyh/entities/Habit.java

```diff
@@ -268,6 +272,64 @@ public String toString() {
                 ", schedule=" + schedule +
                 '}';
     }
-    
+
+    /**
+     * Gets the number of times a habit has been successfully completed
+     * @return number of times a habit has been successfully completed
+     */
+    public int getCompletedCounter() {
+        return completedCounter;
+    }
+
+    /**
+     * Sets the number of times a habit has been successfully completed
+     * @param completedCounter number of times a habit has been successfully completed
+     */
+    public void setCompletedCounter(int completedCounter) {
+        this.completedCounter = completedCounter;
+    }
+
+    /**
+     * Increments the completed counter whenever a habit was successfully completed
+     */
+    public void logCompleted() {completedCounter++;}
```

Comment:
We also need a method to decrement the counter.

Reply:
Done. Method is called undoCompleted()